### PR TITLE
Part of #2966, extending Sprout tests to other epochs.

### PR DESCRIPTION
**Pull request by bitcartel**
Tests updated:
- test_Get
- test_simple_joinsplit_invalidity

### CODE REVIEW

**Comment by bitcartel** [comment]
A single PR for #2966 will be quite large, so breaking it up into smaller pieces.

File: src/test/transaction_tests.cpp [hunk 1]
```diff
@@ -531,9 +529,29 @@ BOOST_AUTO_TEST_CASE(test_simple_joinsplit_invalidity)
...
+        const Consensus::Params& params = Params().GetConsensus();
+        int tmp = params.vUpgrades[Consensus::UPGRADE_OVERWINTER].nActivationHeight;
+        const_cast< Consensus::Params&>(params).vUpgrades[Consensus::UPGRADE_OVERWINTER].nActivationHeight = Consensus::NetworkUpgrade::ALWAYS_ACTIVE;
+        test_simple_joinsplit_invalidity(NetworkUpgradeInfo[Consensus::UPGRADE_OVERWINTER].nBranchId, mtx);
+        const_cast< Consensus::Params&>(params).vUpgrades[Consensus::UPGRADE_OVERWINTER].nActivationHeight = tmp;
+    }
+}
+
+// Parameterized testing over consensus branch ids
+BOOST_DATA_TEST_CASE(test_Get, boost::unit_test::data::xrange(static_cast<int>(Consensus::MAX_NETWORK_UPGRADES)))
```
Comment (str4d) [inline src/test/transaction_tests.cpp]: Ooh, nice feature that I didn't know about :smile:

File: src/test/transaction_tests.cpp [hunk 2]
```diff
@@ -531,9 +529,29 @@ BOOST_AUTO_TEST_CASE(test_simple_joinsplit_invalidity)
...
+    {
+        CMutableTransaction mtx;
+        mtx.fOverwintered = true;
+        mtx.nVersionGroupId = OVERWINTER_VERSION_GROUP_ID;
+        mtx.nVersion = 3;
+        const Consensus::Params& params = Params().GetConsensus();
+        int tmp = params.vUpgrades[Consensus::UPGRADE_OVERWINTER].nActivationHeight;
+        const_cast< Consensus::Params&>(params).vUpgrades[Consensus::UPGRADE_OVERWINTER].nActivationHeight = Consensus::NetworkUpgrade::ALWAYS_ACTIVE;
+        test_simple_joinsplit_invalidity(NetworkUpgradeInfo[Consensus::UPGRADE_OVERWINTER].nBranchId, mtx);
+        const_cast< Consensus::Params&>(params).vUpgrades[Consensus::UPGRADE_OVERWINTER].nActivationHeight = tmp;
```
Comment (str4d) [inline src/test/transaction_tests.cpp]: Use regtest params, and replace the above with:

```cpp
UpdateNetworkUpgradeParameters(Consensus::UPGRADE_OVERWINTER, Consensus::NetworkUpgrade::ALWAYS_ACTIVE);
test_simple_joinsplit_invalidity(NetworkUpgradeInfo[Consensus::UPGRADE_OVERWINTER].nBranchId, mtx);
UpdateNetworkUpgradeParameters(Consensus::UPGRADE_OVERWINTER, Consensus::NetworkUpgrade::NO_ACTIVATION_HEIGHT);
```

Comment (bitcartel) [reply src/test/transaction_tests.cpp]: Updated and pushed.

**Comment by str4d** [comment]
Transient failure.

@zkbot retry

**Pull request merged**